Keyword `description` (#158)
Intelligence-AI/learnjsonschema.com: content/2020-12/meta-data/description.markdown

-----------

This keyword produces the description as the annotation value.

## Explanation

> The `description` keyword in JSON Schema is used to provide additional explanatory information about the schema or its parts. It does not affect data validation but serves as an informative annotation. The value of this keyword must be a string.

Reviewer: ```suggestion
The `description` keyword in JSON Schema is used to provide a human readable description for the schema. It does not affect data validation but serves as an informative annotation. The value of this keyword must be a string.
```

I guess a schema implies its parts?

## Examples

{{<schema `Schema with 'description' keyword`>}}
{
  "$schema": "https://json-schema.org/draft/2020-12/schema",

>   "description": "The age of a person must not be less than zero",

Reviewer: I'm not sure this example is very good. The intention of the `description` keyword is not to express a constraint in prose, but to document the schema. Maybe a description of "The age of a person" makes more sense?

  "type": "number"
}
{{</schema>}}

{{<instance-pass `An instance with a numeric value is valid`>}}
45
{{</instance-pass>}}

{{<instance-annotation>}}
{
  "valid": true,
  "keywordLocation": "/description",
  "instanceLocation": "",

>   "annotation": "The age of a person must not be less than zero"

Reviewer: This annotation is outdated too

Author: Ahh!! Yesterday was quite stressful for me, and I unintentionally became a bit less careful when addressing the suggestions. I apologize for that. I need to be more mindful of your time. I'll make sure this doesn't happen again.

Reviewer: No worries! :)

}
{{</instance-annotation>}}

{{<schema `Schema with logical operators`>}}
{
  "$schema": "https://json-schema.org/draft/2020-12/schema",
  "description": "Personal information of a user",
  "properties": {
    "name": { "type": "string" },
    "age": { "type": "number" }
  },
  "if": {
    "description": "if block",
    "properties": {

>       "age": { "description": "'if' true", "minimum": 18 }

Reviewer: You are adding single quotes around `if` here, but not doing the same for `else` and `then`. Maybe pick one way and stick to it?

Reviewer: Also, this description is similar to the outer description in `if` and same for then/else. Maybe make this specific annotation something like "age"?

Author: Sure!

    }
  },
  "then": {
    "description": "then block",
    "properties": {
      "eligible": { "description": "Eligible", "const": true }
    }
  },
  "else": {
    "description": "else block",
    "properties": {
      "eligible": { "description": "Not eligible", "const": false }
    }
  }
}
{{</schema>}}

{{<instance-pass>}}
{
  "name": "John Doe",
  "age": 25,
  "eligible": true
}
{{</instance-pass>}}

{{<instance-annotation>}}
[
  /// ...
  {
    "valid": true,
    "keywordLocation": "/description",
    "instanceLocation": "",
    "annotation": "Personal information of a user"
  },
  {
    "valid": true,
    "keywordLocation": "/if/description",
    "instanceLocation": "",
    "annotation": "if block"
  },
  {
    "valid": true,
    "keywordLocation": "/if/properties/age/description",
    "instanceLocation": "/age",

>     "annotation": "'Age"

Reviewer: ```suggestion
    "annotation": "Age"
```

There is a weird comma here

>     "annotation": "'if' true"

Reviewer: I think the annotations on this example are now out of the date?

  },
  {
    "valid": true,
    "keywordLocation": "/then/description",
    "instanceLocation": "",
    "annotation": "then block",
  },
  {
    "valid": true,
    "keywordLocation": "/then/properties/eligible/description",
    "instanceLocation": "/eligible",
    "annotation": "then applied"
  },
  // ...
]
{{</instance-annotation>}}

> {{<schema `Schema with multiple annotations for the same instance`>}}

Reviewer: Cool example!

{
  "$schema": "https://json-schema.org/draft/2020-12/schema",

>   "description": "Name of the person should not include special characters",

Reviewer: Like on my other comment before, let's not describe constraints on `description`. Just a statement of the instance. Like "A person name"
